keps/sig-storage/1432-volume-health-monitor/README.md

```diff
@@ -644,38 +680,260 @@ _This section must be completed when targeting alpha to a release._
+   the health monitoring controller sidecar is deployed with the CSI driver.
+   So the only way for a rollout to fail is that deploying the health
+   monitoring controller sidecar with the CSI driver fails. If
+   the health monitoring controller cannot be deployed, no events on volume
+   condition will be reported on PVCs.
+
+   If enabling the `VolumeHealth` feature fails, no event on volume condition will be
+   reported on the pod.
+
+* **What specific metrics should inform a rollback?**
+  An event will be recorded on the PVC when the controller has successfully retrieved an
+  abnormal volume condition from the storage system. When other errors occur in the controller,
+  the errors will also be recorded as events.
+
+  In Kubelet, an event will be recorded on the Pod when Kubelet has successfully retrieved an
+  abnormal volume condition. If the call to `NodeGetVolumeStats` fails for other reasons,
+  an error will be returned and whether this will be logged as an event on the Node is up to
+  the existing Kubelet logic and will not be changed.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and can't do that now.
+  Manual testing will be done.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+  Even if applying deprecation policies, they may still surprise some users.
+  No.
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+  checking if there are objects with field X set) may be a last resort. Avoid
+  logs or events for this purpose.
+
+  An operator can check the metric `csi_sidecar_operations_seconds`,
+  Container Storage Interface operation duration with gRPC error code status
+  total. It is reported from CSI external-health-monitor-controller sidecar.
+  For the health monitor controller sidecar, `csi_sidecar_operations_seconds`
+  will be measuring `ListVolumes` or `GetVolume` RPC.
+  The `csi_sidecar_operations_seconds` metric should be sliced by process after
+  they are aggregated to show metrics for different sidecars.
+
+  In Kubelet, an operator can check whether the feature gate `VolumeHealth`
+  is enabled.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [ ] Metrics
+    - Metric name: csi_sidecar_operations_seconds
+    - [Optional] Aggregation method:
+    - Components exposing the metric:
+      csi-external-health-monitor-controller exposes the `csi_sidecar_operations_seconds` metric.
```

Comment:
it looks like the metric names need to be updated to not be sidecar related

Reply:
I removed the reference for the health monitor agent.  This one is from the controller side so we are keeping it as a CSI sidecar.  It won't have impact on the node.